R_functions/export_estimates.R

```diff
@@ -0,0 +1,491 @@
+    transformed_pe_data <- map(transformed_pe_data,
+                               ~{.$data_grade <- rep("Provisional", nrow(.)); .})
+
+  } else {
+    stop("Invalid value for data_grade. Use 'approved' or 'provisional'.")
+  }
+  #-----------------------------------------------------------------------------------------------------------------#
+  #Combine PE and BSS standardized objects
+  
+  creel_estimates <- list(
+    #table 1
+    stratum_catch = rbind(transformed_pe_data$pe_stratum_catch, transformed_bss_data$bss_stratum_catch),
+    #table 2
+    stratum_effort = rbind(transformed_pe_data$pe_stratum_effort, transformed_bss_data$bss_stratum_effort),
+    #table 3
+    summarized_catch = rbind(transformed_pe_data$pe_summarized_catch, transformed_bss_data$bss_summarized_catch),
+    #table 4
+    summarized_effort = rbind(transformed_pe_data$pe_summarized_effort, transformed_bss_data$bss_summarized_effort)
+  )
+  
+  #combine catch and effort data
+  creel_estimates$stratum <- rbind(creel_estimates$stratum_catch, creel_estimates$stratum_effort)
+  
+  creel_estimates$total <- rbind(creel_estimates$summarized_catch, creel_estimates$summarized_effort)
+  
+  #bring in fishery manager table
+  fishery_manager_trim <- dwg$fishery_manager |>
+    filter(location_type == "Section") |> 
+    select(section_num, catch_area_code, catch_area_description) |>
+    arrange(section_num) |> 
+    distinct() #assumes no duplicate section_num in fishery manager table
+  
+  #join CRC from fishery manager table to estimate tables
+  creel_estimates$stratum <- left_join(creel_estimates$stratum, fishery_manager_trim, by ="section_num")
+   
+  # fishery_manager_total <- as.vector(fishery_manager_trim$catch_area_code)
+  # fishery_manager_total <- paste(fishery_manager_total, collapse = ",")
+  # #join CRCs from all sections for total strata
+  # creel_estimates$total <- creel_estimates$total |> mutate(catch_area_code = fishery_manager_total)
+  
+  #rename value column to estimate_value
+  creel_estimates$stratum <- creel_estimates$stratum |> rename(estimate_value = value)
+  creel_estimates$total <- creel_estimates$total |> rename(estimate_value = value)  
+  
+  # change angler_final capitalization to match creel database lut
+  creel_estimates$stratum <- creel_estimates$stratum |> 
+    mutate(angler_final = case_when(
+      angler_final == "bank" ~ "Bank",
+      angler_final == "boat" ~ "Boat"
+    ))
+  
+  # add period_timestep field to denote yaml model parameters
+  creel_estimates$stratum <- creel_estimates$stratum |> 
+    mutate(period_timestep = case_when(
+      model_type == "PE" ~ params$period_pe,
+      model_type == "BSS" ~ params$period_bss
+      ))
+  
+  creel_estimates$total <- creel_estimates$total |> 
+    mutate(period_timestep = case_when(
+      model_type == "PE" ~ params$period_pe,
+      model_type == "BSS" ~ params$period_bss
+    ))
+  
+  #assign to global env
+  creel_estimates <<- creel_estimates
```

Comment:
I've traded out the <<- operator in this instance for assign() in this commit 5efb51ce0161365ea33d49b4e1b1a101ff368902. I see how <<- is too broad of a brush and how it could make debugging difficult. However, in this case I cannot use return() like I used in the other instance of <<- in generate_analysis_lut.R. Calling return() exits a function at that point and since all of the "load" phase of ETL happens after this it wouldn't work. I can update the creel_estimates object with the transformed/standardized tables produced in the first part of export_estimates() using the assign() function, which does not halt a function. That said, your broader comments about breaking up export_estimates() into smaller chunks might make this a moot point.